src/internal/obj/bucket.go

```diff
@@ -0,0 +1,122 @@
+// Azure calls this a container, because it's not like that word conflicts with anything.
+type Bucket = blob.Bucket
+
+func amazonHTTPClient() (*http.Client, int, error) {
+	advancedConfig := &AmazonAdvancedConfiguration{}
+	if err := cmdutil.Populate(advancedConfig); err != nil {
+		return nil, -1, errors.Wrap(err, "creating amazon http client")
+	}
+	timeout, err := time.ParseDuration(advancedConfig.Timeout)
+	if err != nil {
+		return nil, -1, errors.Wrap(err, "creating amazon http client")
+	}
+	httpClient := &http.Client{Timeout: timeout}
+	if advancedConfig.NoVerifySSL {
+		transport := http.DefaultTransport.(*http.Transport).Clone()
+		transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+		httpClient.Transport = transport
+	}
+	httpClient.Transport = promutil.InstrumentRoundTripper("s3", httpClient.Transport)
+	return httpClient, advancedConfig.Retries, nil
+}
+
+// This seems to be required in order to support disabling ssl verification -- which is needed for EDF testing.
+func amazonSession(ctx context.Context, objURL *ObjectStoreURL) (*session.Session, error) {
+	urlParams, err := url.ParseQuery(objURL.Params)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating amazon session")
+	}
+	endpoint := urlParams.Get("endpoint")
+	// if unset, disableSSL will be false.
+	disableSSL, _ := strconv.ParseBool(urlParams.Get("disableSSL"))
+	region := urlParams.Get("region")
+	httpClient, retries, err := amazonHTTPClient()
+	if err != nil {
+		return nil, errors.Wrap(err, "creating amazon session")
+	}
+	awsConfig := &aws.Config{
+		Region:     aws.String(region),
+		MaxRetries: aws.Int(retries),
+		HTTPClient: httpClient,
+		DisableSSL: aws.Bool(disableSSL),
+		Logger:     log.NewAmazonLogger(ctx),
+	}
+	// Set custom endpoint for a custom deployment.
+	if endpoint != "" {
+		awsConfig.Endpoint = aws.String(endpoint)
+		awsConfig.S3ForcePathStyle = aws.Bool(true)
+	}
+	// Create new session using awsConfig
+	sess, err := session.NewSession(awsConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating amazon session")
+	}
+	return sess, nil
+}
+
+// NewAmazonBucket constructs an amazon client by reading credentials
+// from a mounted AmazonSecret. You may pass "" for bucket in which case it
+// will read the bucket from the secret.
+func NewAmazonBucket(ctx context.Context, objURL *ObjectStoreURL) (*Bucket, error) {
+	// Use or retrieve S3 bucket
+	sess, err := amazonSession(ctx, objURL)
+	if err != nil {
+		return nil, errors.Wrap(err, "amazon bucket")
+	}
+	blobBucket, err := s3blob.OpenBucket(ctx, sess, objURL.Bucket, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "amazon bucket")
+	}
+	return blobBucket, nil
+}
+
+// NewBucket creates a Bucket using the given backend and storage root (for
+// local backends).
+// TODO: Not sure if we want to keep the storage root configuration for
+// non-local deployments. If so, we will need to connect it to the object path
```

Comment:
I think this message can be removed and we can just store everything in the root of the bucket rather than prefixing the root onto the path. Thoughts?